bindgen-tests/tests/expectations/tests/issue-2646.rs

```diff
@@ -0,0 +1,100 @@
+#![allow(dead_code, non_snake_case, non_camel_case_types, non_upper_case_globals)]
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum Plain {
+    Plain1 = 0,
+    Plain2 = 1,
+    Plain3 = 2,
+}
+impl TryFromRaw {
+    pub type TryFromRaw_ctype = ::std::os::raw::c_int;
+    pub const TFR1: TryFromRaw_ctype = -1;
+    pub const TFR2: TryFromRaw_ctype = 5;
+    pub const TFR3: TryFromRaw_ctype = 6;
+}
+#[repr(i32)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum TryFromRaw {
+    TFR1 = -1,
+    TFR2 = 5,
+    TFR3 = 6,
+}
+pub struct TryFromRawError(TryFromRaw_ctype);
+impl TryFromRawError {
+    #[must_use]
+    pub fn value(&self) -> TryFromRaw_ctype {
+        self.0
+    }
+}
+impl std::convert::TryFrom<TryFromRaw_ctype> for TryFromRaw {
+    type Error = TryFromRawError;
+    fn try_from(v: TryFromRaw_ctype) -> Result<Self, Self::Error> {
+        match v {
+            -1 => Ok(TryFromRaw::TFR1),
+            5 => Ok(TryFromRaw::TFR2),
+            6 => Ok(TryFromRaw::TFR3),
+            _ => TryFromRawError(v),
+        }
+    }
+}
+impl FromRawUnchecked {
+    pub type FromRawUnchecked_ctype = ::std::os::raw::c_uint;
```

Comment:
@jbaublitz maybe this should just be `ctype` as it is already inside the `impl` block

Reply:
I just read that having `type` on non-trait `impl` blocks is unstable: https://github.com/rust-lang/rust/issues/8995

So sadly we have to keep this outside the `impl` block.